### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MonicaFM44**

### CODE REVIEW

File: src/api/news/news.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[
```
Comment (ignacioPastor) [inline src/api/news/news.json]: You aren't using this file anymore, are you? Maybe you can just remove it

File: src/app/app.component.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<div>
+    <nav class="navbar sticky-top navbar-light bg-light">
+        <div class="container">
+            <a class="navbar-brand">{{ pageTitle }}</a>
+            <ul class="nav">
+                <li class="nav-item">
+                    <a class="nav-link" [routerLink]="['/home']">Home</a>
```
Comment (ignacioPastor) [inline src/app/app.component.html]: Directives first, always put the most relevant elements at the beginning:
<a [routerLink]="['/home']" class="nav-link">Home</a>

File: src/app/app.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+                </li>
+                <li class="nav-item">
+                    <a class="nav-link" [routerLink]="['/news']">News List</a>
```
Comment (ignacioPastor) [inline src/app/app.component.html]: Directives first

File: src/app/news/INew.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  title: string;
+  description: string;
+  source: string;
+  author: string;
+  date: string;
+  url: string;
+  urlImage: string;
+}
+
+/*status": "ok",
```
Comment (ignacioPastor) [inline src/app/news/INew.ts]: consider master as "production ready", don't let there old comments or personal reminders

File: src/app/news/components/news-detail/news-detail.component.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
+<div class="card" *ngIf="article" style="margin-top: 40px;">
```
Comment (ignacioPastor) [inline src/app/news/components/news-detail/news-detail.component.html]: Try to split style from html, use just inline style if is just to one small thing and you don't want to create a scss file.
Better create here a separate scss file and move there all the style logic of this file

File: src/app/news/components/news-detail/news-detail.component.html [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                    <div class="col-md-3">Source:</div>
+                    <div class="col-md-6">{{ article.source }}</div>
+                </div>
+                <div class="row">
+                    <div class="col-md-3">Description:</div>
+                    <div class=" col-md-8">{{ article.description }}</div>
+                </div>
+                <div class="row ">
+                    <div class="col-md-3 ">URL:</div>
+                    <div class="col-md-8"><a [href]='article.url'>{{ article.url }}</a></div>
```
Comment (ignacioPastor) [inline src/app/news/components/news-detail/news-detail.component.html]: directives first

File: src/app/news/components/news-detail/news-detail.component.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                </div>
+            </div>
+
+            <div class="col-md-6 ">
+                <img class="center-block img-responsive " [style.width.px]="400 " [style.margin.px]="2 " [src]='article.urlImage' [title]='article.title'>
```
Comment (ignacioPastor) [inline src/app/news/components/news-detail/news-detail.component.html]: 1. Double quotes in html
2. Directives first:
```
<img [src]="article.urlImage" [title]="article.title"  class="center-block img-responsive">
```

File: src/app/news/components/news-list/news-list.component.html [hunk 8]
```diff
@@ -0,0 +1,37 @@
+<div class="panel panel-primary" style="margin-top: 40px;">
```
Comment (ignacioPastor) [inline src/app/news/components/news-list/news-list.component.html]: move styles to separate file

File: src/app/news/components/news-list/news-list.component.html [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    <div class="panel-body">
+        <div class="row" style="margin-bottom: 20px;">
+            <div class="col-md-1">Filter by:</div>
+            <div class="col-md-4">
+                <input type="text" [(ngModel)]="listFilter" />
```
Comment (ignacioPastor) [inline src/app/news/components/news-list/news-list.component.html]: directives first

File: src/app/news/components/news-list/news-list.component.html [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            </div>
+        </div>
+        <div class="row">
+            <div class="col-md-6">
+                <h3>Filtered by: {{ listFilter }}</h3>
+            </div>
+        </div>
+        <div class="table-responsive">
+            <table class="table" *ngIf="news && news.length">
```
Comment (ignacioPastor) [inline src/app/news/components/news-list/news-list.component.html]: directives first

File: src/app/news/components/news-list/news-list.component.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+export class NewsListComponent implements OnInit {
+  readonly pageTitle: string = 'News List';
+  filteredNews: INew[];
+  _listFilter: string;
+  get listFilter(): string {
+    return this._listFilter;
+  }
+  set listFilter(value: string) {
+    this._listFilter = value;
+    this.filteredNews = this.filteredNews = this.listFilter ? this.performFilter(this.listFilter) : this.news;
```
Comment (ignacioPastor) [inline src/app/news/components/news-list/news-list.component.ts]: duplicated `this.filteredNews`

File: src/app/news/components/news-list/news-list.component.ts [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  constructor(private _newsService: NewsService) {}
+
+  performFilter(filterBy: string): INew[] {
+    filterBy = filterBy.toLocaleLowerCase();
+    return this.news.filter((myNew: INew) => myNew.title.toLocaleLowerCase().indexOf(filterBy) !== -1);
+  }
+
+  ngOnInit(): void {
+    this._newsService.getNews().subscribe(
+      data1 => {
```
Comment (ignacioPastor) [inline src/app/news/components/news-list/news-list.component.ts]: use semantic names better, has more sense `news`

File: src/app/news/news.module.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import { RouterModule } from '@angular/router';
+import { NewsService } from './news.service';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    FormsModule,
+    RouterModule.forChild([
+      { path: 'news', component: NewsListComponent },
+      { path: 'news/:title', component: NewsDetailComponent }
```
Comment (ignacioPastor) [inline src/app/news/news.module.ts]: Navigate with a title is not the best option, better assign an id and navigate with that. (Titles can contain weird characters and generate unexpected behavior)

File: src/app/news/news.service.ts [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+import { INew } from './INew';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { environment } from '../../environments/environment';
+import { Observable, of } from 'rxjs';
+import { catchError, tap, map } from 'rxjs/operators';
+
+@Injectable()
+export class NewsService {
+  private _newsUrl = 'https://newsapi.org/v2/top-headlines?country=us&apiKey=' + environment.apiKey;
+  // './assets/news.json';
```
Comment (ignacioPastor) [inline src/app/news/news.service.ts]: remove unnecessary comments

File: src/app/news/news.service.ts [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+  // 'https://newsapi.org/v2/top-headlines?country=us&apiKey=' + environment.apiKey;
+
+  constructor(private _http: HttpClient) {}
+
+  getNews(): Observable<INew[]> {
+    return this._http.get(this._newsUrl).pipe(
+      tap(data => console.log('All: ' + JSON.stringify(data))), // tslint:disable-line
```
Comment (ignacioPastor) [inline src/app/news/news.service.ts]: this just to debugging, remove it before moving to master

File: src/app/news/news.service.ts [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+          title: article.title,
+          description: article.description,
+          source: article.source.name,
+          author: article.author,
+          date: article.publishedAt,
+          url: article.url,
+          urlImage: article.urlToImage
+        }));
+      }),
+      catchError((err: HttpErrorResponse) => {
```
Comment (ignacioPastor) [inline src/app/news/news.service.ts]: Unnecessary block statement
```
catchError((err: HttpErrorResponse) => throw err)
```

File: src/app/news/news.service.ts [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+        throw err;
+      })
+    );
+  }
+
+  getArticle(title: string): Observable<INew> {
+    return this.getNews().pipe(map((news: INew[]) => news.find(article => article.title === title)));
+  }
+
+  /*private handleError(err: HttpErrorResponse) {
```
Comment (ignacioPastor) [inline src/app/news/news.service.ts]: Remove this, try to keep the code clean

File: src/assets/news.json [hunk 18]
```diff
@@ -0,0 +1,16 @@
+[
```
Comment (ignacioPastor) [inline src/assets/news.json]: Unused file, remove it

File: src/tslint.json [hunk 19]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (ignacioPastor) [inline src/tslint.json]: duplicate tslint.json? this should be removed (I think)

File: src/api/news/news.json (hunk 1 above)
Comment (MonicaFM44) [reply src/api/news/news.json]: Ok, done

File: src/app/app.component.html (hunk 2 above)
Comment (MonicaFM44) [reply src/app/app.component.html]: Done!

File: src/app/app.component.html (hunk 3 above)
Comment (MonicaFM44) [reply src/app/app.component.html]: Done

File: src/app/news/INew.ts (hunk 4 above)
Comment (MonicaFM44) [reply src/app/news/INew.ts]: Done

File: src/app/news/components/news-detail/news-detail.component.html (hunk 5 above)
Comment (MonicaFM44) [reply src/app/news/components/news-detail/news-detail.component.html]: Done

File: src/app/news/components/news-detail/news-detail.component.html (hunk 6 above)
Comment (MonicaFM44) [reply src/app/news/components/news-detail/news-detail.component.html]: Done

File: src/app/news/components/news-detail/news-detail.component.html (hunk 7 above)
Comment (MonicaFM44) [reply src/app/news/components/news-detail/news-detail.component.html]: Done

File: src/app/news/components/news-list/news-list.component.html (hunk 8 above)
Comment (MonicaFM44) [reply src/app/news/components/news-list/news-list.component.html]: Done

File: src/app/news/components/news-list/news-list.component.html (hunk 9 above)
Comment (MonicaFM44) [reply src/app/news/components/news-list/news-list.component.html]: Done

File: src/app/news/components/news-list/news-list.component.html (hunk 10 above)
Comment (MonicaFM44) [reply src/app/news/components/news-list/news-list.component.html]: Done

File: src/app/news/components/news-list/news-list.component.ts (hunk 11 above)
Comment (MonicaFM44) [reply src/app/news/components/news-list/news-list.component.ts]: Fixed

File: src/app/news/components/news-list/news-list.component.ts (hunk 12 above)
Comment (MonicaFM44) [reply src/app/news/components/news-list/news-list.component.ts]: Changed

File: src/app/news/news.service.ts (hunk 14 above)
Comment (MonicaFM44) [reply src/app/news/news.service.ts]: Done

File: src/app/news/news.service.ts (hunk 15 above)
Comment (MonicaFM44) [reply src/app/news/news.service.ts]: Removed

File: src/app/news/news.service.ts (hunk 17 above)
Comment (MonicaFM44) [reply src/app/news/news.service.ts]: Done

File: src/tslint.json (hunk 19 above)
Comment (MonicaFM44) [reply src/tslint.json]: Yep

File: src/assets/news.json (hunk 18 above)
Comment (MonicaFM44) [reply src/assets/news.json]: I've removed all the assets folder, because is not used anywhere...?

**Pull request merged**